src/cunumeric/sort/sort.cu

```diff
@@ -643,7 +644,7 @@ SegmentMergePiece<legate_type_of<CODE>> merge_all_buffers(
     return result;
   } else {
     // maybe k-way merge is more efficient here...
-    auto exec_policy      = thrust::cuda::par(alloc).on(stream);
+    auto exec_policy      = DEFAULT_POLICY(alloc).on(stream);
```

Comment:
We would need to add synchronization before the cleanup loop in L729 in order to protect buffer destruction.

Reply:
@mfoerste4 can you elaborate why we need to protect buffer destructions?